logemongo

```diff
@@ -30,23 +35,83 @@ GetOptions(\%OPTIONS, qw/
 	password|pass|p=s
 	include|i=s@
 	exclude|x=s@
+	filters|filter=s%
 	debug|D!
 	count|c=i
+	detail!
 /) or pod2usage(2);
 pod2usage(1) if $OPTIONS{help};
 
+# compile all the regexen for SPEED!
+my @INCLUDES = map { qr/$_/ } @{$OPTIONS{include}};
+my @EXCLUDES = map { qr/$_/ } @{$OPTIONS{exclude}};
+
 sub DEBUG {
 	$OPTIONS{debug} ? print "[DEBUG] @_\n" : print "";
 	return;
 }
 
-# compile all the regexen for SPEED!
-my @INCLUDES = map { qr/$_/ } @{$OPTIONS{include}};
-my @EXCLUDES = map { qr/$_/ } @{$OPTIONS{exclude}};
+sub handle {
+	my ($message) = @_;
+	my ($logdata, $payload, %msghash);
```

Comment:
It's not great form to name the `%thing` with "hash" on the end -- the `%` shows that it's a hash.